pennylane_lightning/lightning_gpu/lightning_gpu.py

```diff
@@ -268,15 +268,24 @@ def __init__(  # pylint: disable=too-many-arguments
         self._dp = DevPool()
         self._use_async = use_async
 
-        # Creating the state vector
-        self._mpi_handler = MPIHandler(mpi, mpi_buf_size, len(self.wires), c_dtype)
-
-        self._statevector = self.LightningStateVector(
-            num_wires=len(self.wires),
-            dtype=c_dtype,
-            mpi_handler=self._mpi_handler,
-            use_async=self._use_async,
-        )
+        self._mpi = mpi
```

Comment:
I wonder if we could merge all MPI-specific variables in a single structure. I believe this would fix the codefactor warning. Nothing needs to be done now, though.